pkg/resources/management.cattle.io/v3/setting/validator.go

```diff
@@ -155,33 +340,25 @@ func (a *admitter) validateAgentTLSMode(oldSetting, newSetting v3.Setting) error
 	return nil
 }
 
-func validateDuration(value string) (time.Duration, error) {
-	dur, err := time.ParseDuration(value)
-	if err != nil {
-		return 0, err
-	}
-
-	if dur < 0 {
-		return 0, errors.New("negative duration")
-	}
-
-	return dur, err
-}
-
 func clusterConditionMatches(cluster *v3.Cluster, t v3.ClusterConditionType, status v1.ConditionStatus) bool {
 	for _, cond := range cluster.Status.Conditions {
 		if cond.Type == t && cond.Status == status {
 			return true
 		}
 	}
+
 	return false
 }
 
-func effectiveValue(s v3.Setting) string {
+// effectiveValue returns the effective value of the setting.
+func effectiveValue(s *v3.Setting) string {
```

Comment:
This change is deliberate to avoid unnecessarily dereferencing Setting objects. See above.